src/OpenTelemetry.Exporter.Prometheus.HttpListener/PrometheusHttpListenerOptions.cs

```diff
@@ -10,12 +10,14 @@ namespace OpenTelemetry.Exporter;
 /// </summary>
 public class PrometheusHttpListenerOptions
 {
+    internal const string DefaultScrapeEndpointPath = "/metrics";
+
     private IReadOnlyCollection<string> uriPrefixes = new[] { "http://localhost:9464/" };
 
     /// <summary>
     /// Gets or sets the path to use for the scraping endpoint. Default value: "/metrics".
     /// </summary>
-    public string ScrapeEndpointPath { get; set; } = "/metrics";
+    public string? ScrapeEndpointPath { get; set; } = DefaultScrapeEndpointPath;
```

Comment:
Is the nullable here actually valid?
What is this exporter doing if it doesn't have a valid endpoint?

Reply:
Good question! I went back and forth on this one myself doing review 😄

What happens if it ends up `null`? It gets coalesced back to the default:

* https://github.com/open-telemetry/opentelemetry-dotnet/blob/82b1fbd760eacc4bac2e4f7cf754a36f911e6d77/src/OpenTelemetry.Exporter.Prometheus.HttpListener/PrometheusHttpListener.cs#L31

* https://github.com/open-telemetry/opentelemetry-dotnet/blob/82b1fbd760eacc4bac2e4f7cf754a36f911e6d77/src/OpenTelemetry.Exporter.Prometheus.AspNetCore/PrometheusExporterEndpointRouteBuilderExtensions.cs#L84

* https://github.com/open-telemetry/opentelemetry-dotnet/blob/82b1fbd760eacc4bac2e4f7cf754a36f911e6d77/src/OpenTelemetry.Exporter.Prometheus.AspNetCore/PrometheusExporterApplicationBuilderExtensions.cs#L114

Why not just declare it like this?

```csharp
public string ScrapeEndpointPath { get; set; } = DefaultScrapeEndpointPath;
```

The thing to remember about these annotations is they are just hints for analyzers. There isn't any runtime validation. If it is declared like that, users can still do this...

```csharp
options.ScrapeEndpointPath = null!;
// or (probably something like this is more likely)
string scrapeEndpointPath = null;
options.ScrapeEndpointPath = scrapeEndpointPath;
```

So just because you say "this doesn't allow null" doesn't mean you should stop checking 🤣

The correct way to do it would be something like this...

```csharp
private string scrapeEndpointPath = DefaultScrapeEndpointPath;
public string ScrapeEndpointPath
{ 
   get => scrapeEndpointPath;
   set
   {
       scrapeEndpointPath = value ?? throw new ArgumentNullException(nameof(value));
   }
} 
```

Seemed like a lot more code and there _could be_ a slight benefit to allowing `null` which would be users may clear it using something like [IPostConfigureOptions](https://learn.microsoft.com/dotnet/api/microsoft.extensions.options.ipostconfigureoptions-1?view=net-8.0) if for some reason they want to force the default if some other code tries to change them.

If you really want it to be `string ScrapeEndpointPath` and not `string? ScrapeEndpointPath` LMK I'm happy to go and drop in that code from above as a follow-up 😄